src/jit/lowerxarch.cpp

```diff
@@ -642,15 +643,22 @@ void Lowering::TreeNodeInfoInit(GenTree* tree)
 
         case GT_ARR_OFFSET:
             // This consumes the offset, if any, the arrObj and the effective index,
-            // and produces the flattened offset for this dimension.
-            info->srcCount         = 3;
-            info->dstCount         = 1;
-            info->internalIntCount = 1;
+            // and produces the flattened offset for this dimension.           
+            info->srcCount = 3;
+            info->dstCount = 1;
+
             // we don't want to generate code for this
             if (tree->gtArrOffs.gtOffset->IsIntegralConst(0))
             {
                 MakeSrcContained(tree, tree->gtArrOffs.gtOffset);
             }
+            else
+            {
+                // Here we don't need internal reg to be different from targetReg.
```

Comment:
Done.

---
In reply to: [79939132](https://github.com/dotnet/coreclr/pull/7285#discussion_r79939132) [](ancestors = 79939132)